backends/simplego/dotgeneral_fastpath.go

```diff
@@ -0,0 +1,283 @@
+package simplego
+
+import (
+	"github.com/gomlx/gomlx/pkg/core/dtypes"
+	"github.com/gomlx/gomlx/pkg/core/shapes"
+)
+
+// isContractLastOrder checks if the DotGeneral operands have axes ordered such that
```

Comment:
Per chat in #258 , what about naming this "isMatMulOrder" ? 

Since "ContractLastOrder" is hard to guess what is that order by the reader ("Contract" may imply some contract as opposed to the "contracting", and it is only last for the lhs parameter, but not rhs).

And "MatMul" order may be well understood by many ?

Reply:
Done! Forgot to change this one based on our convo 😅